pkg/advancedapi/advancedApi.go

```diff
@@ -25,65 +25,41 @@ func GetRegisterDocument(resolverClient register.ResolverClient, did string) (*r
 }
 
 // RegisterNewDocument Create and register a new document against the resolver.
-func RegisterNewDocument(resolverClient register.ResolverClient, keyPair *crypto.KeyPair, purpose identity.DidType, name string, override bool) error {
-	did, err := identity.MakeIdentifier(keyPair.PublicKeyBytes)
+func RegisterNewDocument(resolverClient register.ResolverClient, keyPair *crypto.KeyPair, purpose identity.DidType, name string, override bool) (*register.RegisterDocument, error) {
+	registerDocument, issuer, err := NewDocument(purpose, keyPair, name)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	if !override {
-		getDoc, _ := GetRegisterDocument(resolverClient, did)
+		getDoc, _ := GetRegisterDocument(resolverClient, issuer.Did)
 		if getDoc != nil {
-			return nil
+			return getDoc, nil
 		}
 	}
-	issuer, err := register.NewIssuer(did, name)
-	if err != nil {
-		return err
-	}
-	docProof, err := CreateProof(keyPair, issuer, []byte(issuer.Did))
-	if err != nil {
-		return err
-	}
 
-	opts := []register.RegisterDocumentOpts{
-		register.AddRootParams(did, purpose, docProof.Signature, false),
-		register.AddPublicKey(name, keyPair.PublicKeyBase58, false),
-	}
-	registerDocument, errs := register.NewRegisterDocument(opts)
-	if len(errs) != 0 {
-		return fmt.Errorf("error while creating new RegisterDocument: %v", errs)
-	}
-
-	return RegisterUpdatedDocument(resolverClient, registerDocument, keyPair, issuer)
-}
-
-// NewRegisteredIdentity Create and register a new registered identity and its associated register document against the resolver.
-func NewRegisteredIdentity(resolverClient register.ResolverClient, purpose identity.DidType, keyPair *crypto.KeyPair, name string, override bool) (register.RegisteredIdentity, error) {
-	did, err := identity.MakeIdentifier(keyPair.PublicKeyBytes)
+	err = RegisterUpdatedDocument(resolverClient, registerDocument, keyPair, issuer)
 	if err != nil {
 		return nil, err
 	}
 
-	if name == "" {
-		name = identity.MakeName(purpose)
-	}
-	err = validation.ValidateKeyName(name)
-	if err != nil {
-		return nil, err
-	}
+	return registerDocument, nil
+}
 
-	issuer, err := register.NewIssuer(did, name)
+// RegisterNewIdentity Create and register a new registered identity and its associated register document against the resolver.
```

Comment:
Worth writing as `RegisteredIdentity` / `RegisterDocument` (so as not to confuse people about the multiple uses of "register")?
Also - maybe worth indicating what the `RegisterDocument` can be used for (-> to avoid repeated lookups in want to subsequently perform updates.)